vc-verifier/js/src/CredentialsVerifier.js

```diff
@@ -40,22 +41,38 @@ const verifyCredentialSignature = (jwsHeaderAlgoName, publicKey, actualData, sig
     else if (jwsHeaderAlgoName === Constants.JWS_RS256_SIGN_ALGO_CONST) {
         isVerified = crypto.verify('RSA-SHA256', actualData, publicKey, signature);
     }
-    else {
-        throw new Error('Verification algorithm not supported');
-    }
     return isVerified;
 }
 
 export const verifyCredentials = async (credential) => {
     try {
-        preProcessVerifiableCredential(credential);
+        const validationErrorMessage = validate(credential)
+
+        if(isNotNullOrEmpty(validationErrorMessage) && (validationErrorMessage !== Errors.ERROR_VC_EXPIRED)){
+            verificationFailure(validationErrorMessage)
+        }
         const { signature, header } = jws.decode(credential.proof.jws);
         const decodedSignature = Buffer.from(signature, 'base64');
         const publicKeyObject = await getPublicKeyFromVerificationMethod(credential.proof.verificationMethod);
         const canonicalisedCredential = await URDNA2015Canonicalizer(credential);
         const inputData = getJwsSigningInput(header, canonicalisedCredential);
-        return verifyCredentialSignature(header.alg, publicKeyObject, inputData, decodedSignature);
+        const verificationResult = verifyCredentialSignature(header.alg, publicKeyObject, inputData, decodedSignature);
+        if(!verificationResult) {
+            return verificationFailure(`${Errors.SIGNATURE_VERIFICATION_FAILED}`)
+        }
+        return verificationSuccess(validationErrorMessage)
+
     } catch (error) {
-        throw error;
+        return verificationFailure(`${Errors.EXCEPTION_DURING_VERIFICATION}${error.message}`)
     }
-}
\ No newline at end of file
+}
+
+const verificationSuccess = (message) => ({
+    verificationStatus: true,
+    verificationErrorMessage: message ? message: ""
```

Comment:
why verificationSuccess has verificationErrorMessage property?

Reply:
Updated to verificationMessage for both verificationSuccess and verificationFailure so that same object will be returned on both the cases.